modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js

```diff
@@ -12,160 +12,275 @@
+				const {
+					target: {height, width},
+				} = event;
+
+				const imageData = {
+					...{
+						description: '',
+						event,
+						height,
+						title: '',
+						url: '',
+						width,
+					},
+					...item,
+				};
+
+				_dispatchValue({value: imageData}, mergedValues => {
+					onFieldChanged(mergedValues);
+				});
+			});
+			img.src = item.url;
 		}
-	}
-}
-
-ImagePicker.STATE = {
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	errorMessage: Config.string(),
-
-	/**
-	 * @default false
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?bool}
-	 */
-
-	evaluable: Config.bool().value(false),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	fieldName: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	label: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	name: Config.string().required(),
-
-	/**
-	 * @default '000000'
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	predefinedValue: Config.string(),
-
-	/**
-	 * @default false
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?bool}
-	 */
-
-	readOnly: Config.bool().value(false),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof FieldBase
-	 * @type {?(bool|undefined)}
-	 */
-
-	repeatable: Config.bool().value(false),
-
-	/**
-	 * @default false
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(bool|undefined)}
-	 */
-
-	required: Config.bool().value(false),
-
-	/**
-	 * @default true
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(bool|undefined)}
-	 */
-
-	showLabel: Config.bool().value(true),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	spritemap: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	tip: Config.string(),
-
-	/**
-	 * @default undefined
-	 * @instance
-	 * @memberof ImagePicker
-	 * @type {?(string|undefined)}
-	 */
-
-	value: Config.string(),
+	};
+
+	const _handleItemSelectorTriggerClick = event => {
+		event.preventDefault();
+
+		const itemSelectorDialog = new ItemSelectorDialog({
+			eventName: `${portletNamespace}selectDocumentLibrary`,
+			singleSelect: true,
+			url: itemSelectorURL,
+		});
+
+		itemSelectorDialog.on('selectedItemChange', _handleFieldChanged);
+
+		itemSelectorDialog.open();
+	};
+
+	const spritemap =
+		Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg';
+
+	const placeholder = readOnly
+		? ''
+		: Liferay.Language.get('add-image-description');
+
+	let {height = '0', width = '0'} = {
+		...JSON.parse(inputValue || '{}'),
+	};
+
+	height = `${height}px`;
+	width = `${width}px`;
```

Comment:
Instead of overwriting the value of `height` and `width` it is better to do this when using it.

Reply:
Makes sense, removing this overwriting here and interpolating on style usage